### PR TITLE
Range check for non-bounded numeric types

### PR DESCRIPTION
**Pull request by xbauch**
Adds support range checking for signed and unsigned bit-vector types. There is a range check as well.

### CODE REVIEW

File: gnat2goto/driver/range_check.adb [hunk 1]
```diff
@@ -72,7 +72,33 @@ package body Range_Check is
...
+      if Kind (Bound_Type) in I_Bounded_Signedbv_Type | I_Bounded_Floatbv_Type
+      then
+         return Get_Bound_Of_Bounded_Type (Bound_Type, Pos);
+      else
+         declare
+            Type_Width : constant Integer := Get_Width (Bound_Type);
+            Bound_Value : constant Uint :=
+              (if Pos = Bound_Low
+               then Uint_0
+               else UI_From_Int (2 ** Type_Width));
```
Comment (hannes-steffenhagen-diffblue) [inline gnat2goto/driver/range_check.adb]: `2**Type_Width-1`

**Comment by hannes-steffenhagen-diffblue** [comment]
Can we have tests for this?

**Comment by martin-cs** [comment]
Can we also synchronise this with https://github.com/diffblue/gnat2goto/pull/228

**Comment by xbauch** [comment]
> Can we also synchronise this with #228

Yes, I guess #228 makes the first commit redundant (and insufficient).

**Comment by xbauch** [comment]
@hannes-steffenhagen-diffblue I added the test case (at least of the signed type).

File: testsuite/gnat2goto/tests/range_check_non_bounded/test.adb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+procedure Test is
+   type Uint_8 is mod 2**8;
+   Uint_Var : Uint_8 := 5;
+   Int_Var : Integer;
+begin
+   Int_Var := Integer'(Integer (Uint_Var));
```
Comment (hannes-steffenhagen-diffblue) [inline testsuite/gnat2goto/tests/range_check_non_bounded/test.adb]: Why both qualified expression _and_ typecast?

Comment (xbauch) [reply testsuite/gnat2goto/tests/range_check_non_bounded/test.adb]: Qualified expression alone doesn't even parse (expected `integer` type). And typecast alone leads hides the conversion (thus we wouldn't get the range-check).

**Pull request merged**